crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -2109,21 +2130,94 @@ impl Analyzer<'_, '_> {
                     tracker: Default::default(),
                 });
 
-                return Ok(Type::StringMapping(StringMapping {
+                Ok(Type::StringMapping(StringMapping {
                     span,
                     kind: ty.kind.clone(),
                     type_args: TypeParamInstantiation {
                         span: ty.type_args.span,
                         params: vec![arg],
                     },
                     metadata: ty.metadata,
-                }));
+                }))
             }
 
-            _ => {}
+            _ => Ok(Type::StringMapping(ty.clone())),
+        };
+
+        #[allow(clippy::question_mark)]
+        if let Ok(ref ty) = normalized_ty {
+            if let Type::StringMapping(ty) = ty.normalize() {
+                match ty.type_args.params[0].normalize() {
+                    Type::Ref(ref_ty) => {
+                        if let RTsEntityName::Ident(r_ident) = &ref_ty.type_name {
+                            if let Ok(Some(ty_found)) = self.find_type(&r_ident.into()) {
+                                let ty_found = &ty_found.into_iter().map(|v| v.into_owned()).collect::<Vec<Type>>()[0];
+
+                                if let Type::Alias(alias) = ty_found.normalize() {
```

Comment:
Is this logic really required?

Reply:
I'm considering removing `Type::Alias` completely, as it's quite useless type because it requires proper normalization anyway.

Reply:
With the assumption that `expand_top_ref` never returns alias (so it always returns the type of alias) this not needed. I removed type::alias logic and the following still works correctly

```ts
type A = number;
type B = A;
type C = Lowercase<B>; // should error here
```
so I think its ok